test/ec_comp_test.c

```diff
@@ -0,0 +1,2310 @@
+        param_formats_i[param_format_i] < 0 ? "named_curve" : param_formats[param_format_i],
+        comp_formats_i[comp_format_i] < 0 ? "uncompressed" : comp_formats[comp_format_i]);
+
+    b = BIO_new(BIO_s_file());
+    if (b == NULL)
+        goto fail;
+
+    f = fopen(filename, "rb");
+    if (f == NULL) {
+        fprintf(stderr, "# Could not open '%s'\n", filename);
+        return NULL;
+    }
+
+    BIO_set_fp(b, f, BIO_NOCLOSE);
+
+    if (PEM_read_bio_ECPrivateKey(b, &k, NULL, NULL) == 0)
+        goto fail;
+
+    dcfg->asn1_flag  = EC_GROUP_get_asn1_flag(EC_KEY_get0_group(k));
+    dcfg->point_form = EC_GROUP_get_point_conversion_form(EC_KEY_get0_group(k));
+
+    if (param_formats_i[param_format_i] >= 0)
+        EC_KEY_set_asn1_flag(k, param_formats_i[param_format_i]);
+    if (comp_formats_i[comp_format_i] >= 0)
+        EC_KEY_set_conv_form(k, comp_formats_i[comp_format_i]);
+
+    BIO_free(b);
+    fclose(f);
+    return k;
+
+fail:
+    EC_KEY_free(k);
+    BIO_free(b);
+    fclose(f);
+    return NULL;
+    //return generate_new_key(curve_i, asn1_flag, point_form);
+}
+
+static int should_skip(size_t curve_i, size_t param_format_i)
+{
+    if (known_curves[curve_i].nid == nid_Oakley_EC2N_3)
+        return 1;
+    if (known_curves[curve_i].nid == nid_Oakley_EC2N_4)
+        return 1;
+    return 0;
+}
+
+#ifndef GENERATE_KNOWN_RESULTS
+static int verify_known_result(size_t curve_i, size_t param_format_i, size_t comp_format_i, size_t key_src, struct default_config_st *dcfg)
+{
+    size_t i;
+
+    for (i=0; i<OSSL_NELEM(known_results); ++i) {
+        if (strcmp(known_results[i].curve_name, known_curve_names[curve_i]) != 0)
+            continue;
+        if (param_formats_i[param_format_i] != known_results[i].param_format)
+            continue;
+        if (comp_formats_i[comp_format_i] != known_results[i].comp_format)
+            continue;
+        if (key_src != known_results[i].key_src)
+            continue;
+
+        break;
+    }
+
+    if (i >= OSSL_NELEM(known_results)) {
+        fprintf(stderr, "# could not identify known result\n");
+        return 0;
+    }
+
+    if (known_results[i].res_param_format != dcfg->asn1_flag) {
+        fprintf(stderr, "# EC parameter format mismatch\n");
+        return 0;
+    }
+
+    if (known_results[i].res_comp_format != dcfg->point_form) {
+        fprintf(stderr, "# EC compression format mismatch\n");
+        return 0;
+    }
+
+    return 1;
+}
+#endif
+
+/*
+ * curve_i: Index into known_curves.
+ * param_format_i: 0=Named parameters, 1=Explicit parameters.
+ * comp_format_i: Index into comp_formats(_i).
+ *   If this exceeds the size of comp_formats, does not configure compression
+ *   format.
+ * key_src: 0=Deserialize an existing key. 1=Generate a new key.
+ */
+static int comp_test_actual(size_t curve_i,
+                            size_t param_format_i,
+                            size_t comp_format_i,
+                            size_t key_src)
```

Comment:
also handle input arguments `int key_or_params`

We want to repeat this test also for an `EC_KEY` object that does not hold a key inside, but only the parameters (the `EC_GROUP`), basically the result of deserializing a file with only EC_PARAMETERS in it.